src/test/ui/type-alias-impl-trait/bound_reduction2.stderr

```diff
@@ -1,5 +1,14 @@
+error[E0277]: the trait bound `T: TraitWithAssoc` is not satisfied
+  --> $DIR/bound_reduction2.rs:10:1
+   |
+LL | type Foo<V> = impl Trait<V>;
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `TraitWithAssoc` is not implemented for `T`
+...
+LL | fn foo_desugared<T: TraitWithAssoc>(_: T) -> Foo<T::Assoc> {
+   |                  -- help: consider further restricting this bound: `T: TraitWithAssoc +`
```

Comment:
This error looks pretty wrong though, doesn't it?

Reply:
Yeah, but it makes sense in the context.
It's the error you get if the other errors don't prevent the `impl Trait` from getting the incompatible concrete type, because of the type parameter that is now in the wrong `ParamEnv`, which doesn't have the original bound.

This is pre-existing, it was just hidden by the bug https://github.com/rust-lang/rust/pull/66963 fixes.